### PR TITLE
Merge 3.5 final into master

### PR DESCRIPTION
**Pull request by jkmassel**
Merges all 3.5 changes into `master`.

### CODE REVIEW

File: RELEASE-NOTES.txt [hunk 1]
```diff
@@ -1,6 +1,13 @@
+3.6
```
Comment (loremattei) [inline RELEASE-NOTES.txt]: This is kind of unexpected merging the `release/3.5` branch. 
Do you know why it's here?

Comment (jkmassel) [reply RELEASE-NOTES.txt]: Yeah – https://github.com/woocommerce/woocommerce-ios/commit/238063d64347cb110407e0d092192f2bb2055d53 was merged prior to the last code freeze (on Jan 22, code freeze was Jan 27), and introduced the `3.6` notes early.

Comment (loremattei) [reply RELEASE-NOTES.txt]: So the feature was under a feature flag and it's been released in 3.6?

Comment (jkmassel) [reply RELEASE-NOTES.txt]: It wasn't under a feature flag, but it _was_ released in 3.6 and is in the 3.6 release notes at `p2JRYi-4qp` :)

Comment (loremattei) [reply RELEASE-NOTES.txt]: 👍

**Pull request merged**